x-pack/plugins/security_solution/public/resolver/view/panels/panel_content_related_list.tsx

```diff
@@ -0,0 +1,240 @@
+  eventCategory: string;
+  name: string;
+  entityId: string;
+  setQueryParams: () => void;
+}
+
+const DisplayList = memo(function DisplayList({
+  crumbs,
+  matchingEventEntries,
+}: {
+  crumbs: Array<{ text: string | JSX.Element; onClick: () => void }>;
+  matchingEventEntries: MatchingEventEntry[];
+}) {
+  return (
+    <>
+      <StyledBreadcrumbs breadcrumbs={crumbs} />
+      <EuiSpacer size="l" />
+      <>
+        {matchingEventEntries.map((eventView, index) => {
+          return (
+            <Fragment key={index}>
+              <EuiText>
+                <BoldCode>
+                  <FormattedMessage
+                    id="xpack.securitySolution.enpoint.resolver.panel.relatedEventDetail.categoryAndType"
+                    values={{
+                      category: eventView.eventCategory,
+                      eventType: eventView.eventType,
+                    }}
+                    defaultMessage="{category} {eventType}"
+                  />
+                </BoldCode>
+                <FormattedMessage
+                  id="xpack.securitySolution.enpoint.resolver.panel.relatedEventDetail.atTime"
+                  values={{ date: eventView.formattedDate }}
+                  defaultMessage="@ {date}"
+                />
+              </EuiText>
+              <EuiSpacer size="xs" />
+              <EuiButtonEmpty onClick={eventView.setQueryParams}>{eventView.name}</EuiButtonEmpty>
+              {index === matchingEventEntries.length - 1 ? null : <EuiHorizontalRule margin="m" />}
+            </Fragment>
+          );
+        })}
+      </>
+    </>
+  );
+});
+
+export const ProcessEventListNarrowedByType = memo(function ProcessEventListNarrowedByType({
+  processEvent,
+  eventType,
+  relatedStats,
+  pushToQueryParams,
+}: {
+  processEvent: ResolverEvent;
+  pushToQueryParams: (arg0: CrumbInfo) => unknown;
+  eventType: string;
+  relatedStats: ResolverNodeStats;
+}) {
+  const processName = processEvent && event.eventName(processEvent);
+  const processEntityId = event.entityId(processEvent);
+  const totalCount = Object.values(relatedStats.events.byCategory).reduce(
+    (sum, val) => sum + val,
+    0
+  );
+  const eventsString = i18n.translate(
+    'xpack.securitySolution.enpoint.resolver.panel.processEventListByType.events',
+    {
+      defaultMessage: 'Events',
+    }
+  );
+  const waitingString = i18n.translate(
+    'xpack.securitySolution.enpoint.resolver.panel.processEventListByType.wait',
+    {
+      defaultMessage: 'Waiting For Events...',
+    }
+  );
+
+  const relatedsReadyMap = useSelector(selectors.relatedEventsReady);
+  const relatedsReady = relatedsReadyMap.get(processEntityId);
+
+  const relatedEventsForThisProcess = useSelector(selectors.relatedEventsByEntityId).get(
+    processEntityId
+  );
+  const dispatch = useResolverDispatch();
+
+  useEffect(() => {
+    if (typeof relatedsReady === 'undefined') {
+      dispatch({
+        type: 'userRequestedRelatedEventData',

```

Comment:
re: our discussion. Could you change this to a new action that starts w/ 'app'? ty

Reply:
https://github.com/elastic/kibana/pull/69098/commits/1058e680726ef3f49b925e610a67cf750a43c827